src/NServiceBus.AzureStoragePersistence/SagaPersisters/Azure/SecondaryIndeces/SecondaryIndexPersister.cs

```diff
@@ -0,0 +1,196 @@
+﻿namespace NServiceBus.SagaPersisters.Azure.SecondaryIndeces
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.IO;
+    using System.Linq.Expressions;
+    using System.Net;
+    using Microsoft.WindowsAzure.Storage;
+    using Microsoft.WindowsAzure.Storage.Table;
+    using Newtonsoft.Json;
+    using NServiceBus.Saga;
+
+    public class SecondaryIndexPersister
+    {
+        readonly Func<Type, CloudTable> getTableForSaga;
+        readonly ScanForSaga scanner;
+        readonly Action<IContainSagaData> persist;
+        readonly ConcurrentDictionary<Type, IndexDefintion> sagaToIndex = new ConcurrentDictionary<Type, IndexDefintion>();
+
+        const int LRUCapacity = 1000;
+        readonly LRUCache<PartitionRowKeyTuple, Guid> cache = new LRUCache<PartitionRowKeyTuple, Guid>(LRUCapacity);
+
+        public SecondaryIndexPersister(Func<Type, CloudTable> getTableForSaga, ScanForSaga scanner, Action<IContainSagaData> persist)
+        {
+            this.getTableForSaga = getTableForSaga;
+            this.scanner = scanner;
+            this.persist = persist;
+        }
+
+        public delegate Guid? ScanForSaga(Type sagaType, string propertyName, object propertyValue);
+
+        public void Insert(IContainSagaData sagaData)
+        {
+            var sagaType = sagaData.GetType();
+            var table = getTableForSaga(sagaType);
+            var key = GetIndexKey(sagaData, sagaType);
+
+            var entity = new SecondaryIndexTableEntity
+            {
+                SagaId = sagaData.Id,
+                SagaData = SagaDataSerializer.SerializeSagaData(sagaData),
```

Comment:
Table entity properties can contain a maximum of 32000 characters (64K, UTF16 encoded), best to add compression here?